python/mxnet/initializer.py

```diff
@@ -91,11 +91,28 @@ def _init_weight(self, _, arr):
 
 
 class Xavier(Initializer):
-    """Initialize the weight with Xavier initialization scheme."""
+    """Initialize the weight with Xavier or similar initialization scheme.
+
+    Parameters
+    ----------
+    rnd_type: str, optional
+        Use ```guassian``` or ```uniform``` to init
+
+    magnitude: float, optional
+        scale of random number range
+    """
+    def __init__(self, rnd_type="uniform", magnitude=6):
+        self.rnd_type = rnd_type
+        self.magnitude = magnitude
+
 
     def _init_weight(self, _, arr):
         shape = arr.shape
         fan_in, fan_out = np.prod(shape[1:]), shape[0]
-        scale = np.sqrt(3. / (fan_in + fan_out))
-        random.uniform(-scale, scale, out=arr)
-
+        scale = np.sqrt(self.magnitude / (fan_in + fan_out))
```

Comment:
I think the name `Xavier` is well established so far. Caffe only distinguishes between `Xavier` and `MSRA`, but the only real difference is the magnitude for each scheme. For both variants they then support different regularizations `/in` `/out` `2/(in+out)`.